Utils/Extensions/StringExtensions.swift

```diff
@@ -40,4 +44,105 @@ extension String {
+        return hash?.base64EncodedString
+    }
+
+    private func performEncryptionOperation(operation: Int, key: NSString, iv: NSMutableData, data: NSData) -> NSData? {
+        let keyData: NSData! = (key as NSString).dataUsingEncoding(NSUTF8StringEncoding) as NSData!
+        let keyBytes         = UnsafePointer<Void>(keyData.bytes)
+
+        let dataLength    = data.length
+        let dataBytes     = UnsafePointer<Void>(data.bytes)
+
+        let cryptData    = NSMutableData(length: Int(dataLength) + kCCBlockSizeAES128)
+        var cryptPointer = UnsafeMutablePointer<Void>(cryptData!.mutableBytes)
+        let cryptLength  = cryptData!.length
+
+        let keyLength              = kCCKeySizeAES256
+        let operation: CCOperation = UInt32(operation)
+        // Note 128 is the block length here, not the key length
+        let algoritm:  CCAlgorithm = UInt32(kCCAlgorithmAES128)
+        let options:   CCOptions   = UInt32(kCCModeCBC + kCCOptionPKCS7Padding)
+
+        var numBytesEncrypted: Int = 0
+        var cryptStatus = CCCrypt(operation,
+            algoritm,
+            options,
+            keyBytes, keyLength,
+            iv.bytes,
+            dataBytes, dataLength,
+            cryptPointer, cryptLength,
+            &numBytesEncrypted)
+
+        let success: CCCryptorStatus = Int32(kCCSuccess)
+        if cryptStatus == success {
+            return cryptData!.subdataWithRange(NSRange(location: 0, length: numBytesEncrypted))
+        }
+
+        return nil;
+    }
+
+    private func decodeData(data: NSData) -> (hmac: String, iv: NSMutableData, payload: String) {
+        let decoder = NSKeyedUnarchiver(forReadingWithData: data)
+        let hmac = decoder.decodeObjectForKey(HmacKey) as! String
+        let iv = decoder.decodeObjectForKey(IvKey) as! NSMutableData
+        let payload = decoder.decodeObjectForKey(PayloadKey) as! String
+        decoder.finishDecoding()
+
+        return (hmac, iv, payload)
+    }
+
+    private func encodeData(hmac: String, iv: NSMutableData, payload: String) -> NSData {
+        let data = NSMutableData()
+        let encoder = NSKeyedArchiver(forWritingWithMutableData: data)
+        encoder.encodeObject(hmac, forKey: HmacKey)
+        encoder.encodeObject(iv, forKey: IvKey)
+        encoder.encodeObject(payload, forKey: PayloadKey)
+        encoder.finishEncoding()
+        return data
+    }
+
+    public func AES256EncryptWithKey(keyString: String) -> String? {
+        let data: NSData! = dataUsingEncoding(NSUTF8StringEncoding) as NSData!
+        let iv = NSMutableData(length: 16)
+        let err = SecRandomCopyBytes(kSecRandomDefault, 16, UnsafeMutablePointer(iv!.mutableBytes))
+        if err != 0 {
+            return nil
+        }
+
+        if let encrypted = performEncryptionOperation(kCCEncrypt, key: keyString, iv: iv!, data: data) {
+            let payload = encrypted.base64EncodedString
+            let hmac = shaw256(keyString + "Salt", payload: payload)
```

Comment:
I don't know if adding "Salt", a fixed string, to the key actually adds anything from a crypto point of view. My gut feeling says it is meaningless. We should ask one of our in-house crypto gurus I think.